Small improvements

TweakM opened the pull request:
This PR adds three things:
1. Correct spelling error in mrtg.cfg
2. Only run indexmaker if the index file does not exist. Without this, each container restart the index will be regenerated and appended to the existing file.
3. Add the capability to specify a USERID and GROUPID for lighttpd.

TweakM committed:
Correct spelling error in mrtg.cfg

TweakM committed:
Only run indexmaker if the index file does not exist. Without this, each container restart the index will be regenerated and appended to the existing file.

TweakM committed:
Add the capability to specify a USERID and GROUPID for lighttpd.

fboaventura commented on files/mrtg.sh:
```diff
@@ -52,7 +55,12 @@ sleep 2
+
+# Only run indexmaker if the index file does not exist. Without this, each container restart the index will be regenerated and appended to the existing file.
+if [ ! -e "${WEBDIR}/index.html" ]; then
+   /usr/bin/indexmaker --columns=1 ${MRTGCFG} -rrdviewer=${PATHPREFIX}/cgi-bin/14all.cgi --icondir=/ --prefix=${PATHPREFIX}/ >> ${WEBDIR}/index.html
```
@TweakM I remember having a reason to re-run the `indexmaker` on every reboot. I won't be able to validate it now, but if I'm not mistaken, we need this to account for new assets added to the list.

If that is the case, we may then regenerate the index on each run and overwrite the file. Can you please, validate?

TweakM replied:
@fboaventura You are right, I did not take the use case of adding new assets into account. Due to the append bug it still needs a fix though. I'll see if I can create an update that satisfies both use cases (my use case being a manually modified index.html and/or a manually modified cfg file).

fboaventura replied:
@TweakM replacing the `>>` by `>` fixes the ever-increasing `index.html`.

I can think of two quick paths: 1. Storing the number of assets in a file and checking this file on each run to decide if it will execute `indexmaker` or not, and 2. Have a variable to determine if it will be updated or not.

Depending on the customization depth, we may have new variables for `indexmaker` [options](https://manpages.ubuntu.com/manpages/xenial/man1/indexmaker.1.html) or even the templating with [configmaker](https://oss.oetiker.ch/mrtg/doc/cfgmaker.en.html#Details_on_Templates). 

Is this enough? If not, can you share some of your customizations?

TweakM replied:
@fboaventura I have implemented the second path (a variable that determines if the `index.html` should be regenerated or not). I have also included an environment variable to specify extra `indexmaker ` options. The documentation has also been updated with my latest [commit](https://github.com/fboaventura/dckr-mrtg/pull/13/commits/5d15dcf85f6dd45d9f724096b2e4b4055e544013). My customizations are manual edits to the `index.html` that cannot be generated by `indexmaker`.
Hopefully this satisfies all your questions and requests.

fboaventura commented on Dockerfile:
```diff
@@ -3,9 +3,11 @@ FROM alpine:3.16
 ENV HOSTS "public:localhost"
+ENV USERID "100"
+ENV GROUPID "101"
 
```
@TweakM Can you please, just set the defaults to a higher ID? Since being below 500, it may clash with the host OS managed.

I, personally, would go with 1000 or 999. :)

fboaventura replied:
Or are these IDs created automatically by lighttpd?

TweakM replied:
@fboaventura I agree that one should normally go for 1000 for the IDs. To prevent breaking existing running containers when they are upgraded I kept the values that the current setup has by default:
/etc/passwd:
`lighttpd:x:100:101:lighttpd:/var/www/localhost/htdocs:/sbin/nologin
`
New users can safely use the values they need. Do you agree?

fboaventura reviewed (changes requested):
@TweakM thank you very much for your contribution!

I've left some comments, can you please address them?

TweakM committed:
Added the option to not regenerate the index.html file, applicable when custom/manual changes to this file have been made
Added the ability to specify additional options for `indexmaker` (allowing more customizations)
Updated documentation

TweakM committed:
Update VERSION in accordance with the changes

fboaventura reviewed (approved):
Great work! LGTM! ;)

TweakM commented:
@fboaventura : Thank you!
Will you also update the images on Docker Hub? (and maybe also the documentation there). That will allow everyone to use the improvements.

fboaventura commented:
@TweakM, the image is already published. I'm dealing with a known issue with the auto-build that fails before publishing the documentation.
